src/readbin/symbols.ml

```diff
@@ -66,4 +66,20 @@ let read ?demangle ~filename arch base  : string table =
             let mem = Memory.view ~from ~words base |> ok_exn in
             let name = demangle name in
             Table.add syms mem name |> ok_exn
-          with exn -> syms))
+          with _exn -> syms))
+
+let read_addrset filename : Addr.Set.t =
+  Addr.Set.t_of_sexp @@ Sexp.load_sexp filename
+
+let write_addrset ?filename (addr_set : Addr.Set.t) : unit =
```

Comment:
ok, rewrite all io utilities in a such way, that they accept `channel` and let the caller side to decide where to print and what to close.